src/deploy/scope_subscription.ts

```diff
@@ -59,29 +61,36 @@ export async function DeploySubscriptionScope(azPath: string, region: string, te
     }
 
     // validate the deployment
-    core.info("Validating template...")
-    var code = await exec(`"${azPath}" deployment sub validate ${azDeployParameters} -o json`, [], validateOptions);
-    if (deploymentMode === "validate" && code != 0) {
-        throw new Error("Template validation failed.")
-    } else if (code != 0) {
-        core.warning("Template validation failed.")
+    if (!whatIf) {
+        core.info("Validating template...")
+        var code = await exec(`"${azPath}" deployment sub validate ${azDeployParameters} -o json`, [], validateOptions);
+        if (deploymentMode === "validate" && code != 0) {
```

Comment:
string comparison should be case insensitive everywhere. What is the general pattern followed in actions?
The unexpected side effect of doing case sensitive comparison here is that because of a case typo by the user, we would go and deploy their template rather than validate.

Reply:
Done